### DISCUSSION TITLE
How to delete a polymorphic row when deleting a model?

### DISCUSSION THREAD

**Discussion by lucassimines** [sequelize/sequelize#14411] [Help & Questions] [answered]
Hello guys, I have a Category model that has a polymorphic table called "categorizables"

I have model called `Product` that has association with the `Category` model. But when I delete a product, the categorizable table still keeps the record of `product_id category_id` how can I delete this relation row when deleting a product?

**Comment by ephys** [comment, marked as answer]
If I'm not mistaken, a polymorphic row is a row that can reference different tables?
If that's the case, that is not supported, as we rely on `ON DELETE CASCADE` to delete the row from the through table, and such a row would not have a foreign key.

You need to delete it manually
